include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+    SDL_GPU_BLENDOP_MIN,
+    SDL_GPU_BLENDOP_MAX
+} SDL_GpuBlendOp;
+
+typedef enum SDL_GpuBlendFactor
+{
+    SDL_GPU_BLENDFACTOR_ZERO,
+    SDL_GPU_BLENDFACTOR_ONE,
+    SDL_GPU_BLENDFACTOR_SRC_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_SRC_COLOR,
+    SDL_GPU_BLENDFACTOR_DST_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_DST_COLOR,
+    SDL_GPU_BLENDFACTOR_SRC_ALPHA,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_SRC_ALPHA,
+    SDL_GPU_BLENDFACTOR_DST_ALPHA,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_DST_ALPHA,
+    SDL_GPU_BLENDFACTOR_CONSTANT_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_CONSTANT_COLOR,
+    SDL_GPU_BLENDFACTOR_SRC_ALPHA_SATURATE
+} SDL_GpuBlendFactor;
+
+typedef enum SDL_GpuColorComponentFlagBits
+{
+    SDL_GPU_COLORCOMPONENT_R_BIT = 0x00000001,
+    SDL_GPU_COLORCOMPONENT_G_BIT = 0x00000002,
+    SDL_GPU_COLORCOMPONENT_B_BIT = 0x00000004,
+    SDL_GPU_COLORCOMPONENT_A_BIT = 0x00000008
+} SDL_GpuColorComponentFlagBits;
+
+typedef Uint32 SDL_GpuColorComponentFlags;
+
+typedef enum SDL_GpuFilter
+{
+    SDL_GPU_FILTER_NEAREST,
+    SDL_GPU_FILTER_LINEAR
+} SDL_GpuFilter;
+
+typedef enum SDL_GpuSamplerMipmapMode
+{
+    SDL_GPU_SAMPLERMIPMAPMODE_NEAREST,
+    SDL_GPU_SAMPLERMIPMAPMODE_LINEAR
+} SDL_GpuSamplerMipmapMode;
+
+typedef enum SDL_GpuSamplerAddressMode
+{
+    SDL_GPU_SAMPLERADDRESSMODE_REPEAT,
+    SDL_GPU_SAMPLERADDRESSMODE_MIRRORED_REPEAT,
+    SDL_GPU_SAMPLERADDRESSMODE_CLAMP_TO_EDGE
+} SDL_GpuSamplerAddressMode;
+
+/*
+ * VSYNC:
+ *   Waits for vblank before presenting.
+ *   If there is a pending image to present, the new image is enqueued for presentation.
+ *   Disallows tearing at the cost of visual latency.
+ *   When using this present mode, AcquireSwapchainTexture will block if too many frames are in flight.
+ * IMMEDIATE:
+ *   Immediately presents.
+ *   Lowest latency option, but tearing may occur.
+ *   When using this mode, AcquireSwapchainTexture will return NULL if too many frames are in flight.
+ * MAILBOX:
+ *   Waits for vblank before presenting. No tearing is possible.
+ *   If there is a pending image to present, the pending image is replaced by the new image.
+ *   Similar to VSYNC, but with reduced visual latency.
+ *   When using this mode, AcquireSwapchainTexture will return NULL if too many frames are in flight.
+ */
+typedef enum SDL_GpuPresentMode
+{
+    SDL_GPU_PRESENTMODE_VSYNC,
+    SDL_GPU_PRESENTMODE_IMMEDIATE,
+    SDL_GPU_PRESENTMODE_MAILBOX
+} SDL_GpuPresentMode;
+
+/*
+ * SDR:
+ *   B8G8R8A8 or R8G8B8A8 swapchain. Pixel values are in nonlinear sRGB encoding. Blends raw pixel values.
+ * SDR_LINEAR:
+ *   B8G8R8A8_SRGB or R8G8B8A8_SRGB swapchain. Pixel values are in nonlinear sRGB encoding. Blends in linear space.
+ * HDR_EXTENDED_LINEAR:
+ *   R16G16B16A16_SFLOAT swapchain. Pixel values are in extended linear encoding. Blends in linear space.
+ * HDR10_ST2048:
+ *   A2R10G10B10 or A2B10G10R10 swapchain. Pixel values are in PQ ST2048 encoding. Blends raw pixel values. (TODO: verify this)
```

Comment:
Blending raw pixels doesn't make sense here. Typically when this format is used an application will create an offscreen render target in linear space and then do final shader conversion from linear to PQ output. At least on Windows it's as efficient or more efficient to use `HDR_EXTENDED_LINEAR` and let the system do PQ conversion if needed. I started adding support for this to the render API and dropped it because of this.

Reply:
Is this just for picking the swapchain colorspace?

I imagine most swapchains are opaque. Not sure why blending is talked about here? Steam Deck/Gamescope always blends in linear space for a submitted swapchain buffer.

We definitely prefer HDR10/PQ on Steam Deck fwiw.

Reply:
Yeah, this is exclusively for the swapchain - the blending part I think only applies when the SDL_Window in question has the transparency flag set, otherwise we just use VK_COMPOSITE_ALPHA_OPAQUE_BIT_KHR.

Reply:
I'm gonna be honest, I barely understand how the HDR swapchain stuff works at all, I just wired it up as best I could. So any clarifications to this documentation are more than welcome.

Reply:
Blending on the swapchain doesn't work in D3D12, if I remember correctly. I would just drop the blending comments here. I would also recommend calling this `SDL_GpuSwapchainColorspace` for clarity and consistency with the rest of SDL.

Reply:
The problem is it's not just a colorspace, it also defines the backing texture format. These used to be separate enums but 99.9% of the combinations are completely invalid and the client shouldn't have to worry about minutiae like whether the swapchain format on the OS is BGRA or RGBA.

Reply:
Okay, that's fine.